### PR TITLE
 Remove applications that depend on python2Packages (Part 2)

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change
https://github.com/NixOS/nixpkgs/issues/148779

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mweinelt** [comment]
@thoughtpolice `opae` still fails to build, PTAL.

**Comment by mweinelt** [comment]
- blink: @pSub 
- curaByDagoma: @tiramiseb 
- neap: @romildo 
- metamorphose2: @RamKromberg 
- lumpy: @jbedo 
- cvs2svn: @makefu 
- gitinspector: @bjornfor 
- ino: @antono 
- getmail: @7c6f434c 
- displaycal: @MarcWeber

**Comment by tiramiseb** [comment]
You can remove curaByDagoma. I'm not using it anymore and I bet nobody else ever did.

**Comment by makefu** [comment]
thanks for the cleanup!

**Review by romildo — APPROVED** [review]
neap can be removed. It seems that it is not being developed anymore. If sometime it get ported to python 3, it may be added again to nixpkgs.

**Comment by 7c6f434c** [comment]
Not aware of any benefit of getmail over getmail6 now that I look at this pair in Nixpkgs…

**Comment by 7c6f434c** [comment]
Tailor and csv2svn (that is also cvs2git) seem useful and trivial to fix via PyPy2. #154752

**Comment by dotlambda** [comment]
> 

Should we turn getmail into an alias for getmail6 or suggest using getmail6 in the appropriate throw?

**Comment by 7c6f434c** [comment]
Upstream promises to be drop-in replacement, so I guess alias could be fine.

**Comment by mweinelt** [comment]
> > Not aware of any benefit of getmail over getmail6 now that I look at this pair in Nixpkgs…
> 
> Should we turn getmail into an alias for getmail6 or suggest using getmail6 in the appropriate throw?

I did the latter.

**Comment by mweinelt** [comment]
Result of `nixpkgs-review pr 154697` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>13 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.hardkernel_4_14.perf</li>
    <li>linuxKernel.packages.hardkernel_4_14.systemtap</li>
    <li>linuxKernel.packages.linux_4_14.perf</li>
    <li>linuxKernel.packages.linux_4_14_hardened.perf</li>
    <li>linuxKernel.packages.linux_4_19.perf</li>
    <li>linuxKernel.packages.linux_4_19_hardened.perf</li>
    <li>linuxKernel.packages.linux_4_4.perf</li>
    <li>linuxKernel.packages.linux_4_9.perf</li>
    <li>linuxPackages_4_14_hardened.perf</li>
    <li>linuxPackages_4_19_hardened.perf</li>
    <li>linuxPackages_hardkernel_latest.perf</li>
    <li>linuxPackages_hardkernel_latest.systemtap</li>
    <li>tlspool</li>
  </ul>
</details>
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>5 packages failed to build:</summary>
  <ul>
    <li>opae</li>
    <li>python310Packages.cartopy</li>
    <li>python310Packages.fiona</li>
    <li>python310Packages.geopandas</li>
    <li>python310Packages.osmnx</li>
  </ul>
</details>
<details>
  <summary>48 packages built:</summary>
  <ul>
    <li>cargo-flamegraph</li>
    <li>digikam</li>
    <li>enblend-enfuse</li>
    <li>gdal_2</li>
    <li>git-crecord</li>
    <li>hugin</li>
    <li>linuxKernel.packages.linux_4_14.systemtap</li>
    <li>linuxPackages_4_14_hardened.systemtap (linuxKernel.packages.linux_4_14_hardened.systemtap)</li>
    <li>linuxKernel.packages.linux_4_19.systemtap</li>
    <li>linuxPackages_4_19_hardened.systemtap (linuxKernel.packages.linux_4_19_hardened.systemtap)</li>
    <li>linuxKernel.packages.linux_4_4.systemtap</li>
    <li>linuxKernel.packages.linux_4_9.systemtap</li>
    <li>linuxPackages.perf (linuxKernel.packages.linux_5_10.perf)</li>
    <li>linuxPackages.systemtap (linuxKernel.packages.linux_5_10.systemtap)</li>
    <li>linuxPackages_hardened.perf (linuxPackages_5_10_hardened.perf)</li>
    <li>linuxPackages_hardened.systemtap (linuxPackages_5_10_hardened.systemtap)</li>
    <li>linuxKernel.packages.linux_5_15.perf</li>
    <li>linuxKernel.packages.linux_5_15.systemtap</li>
    <li>linuxPackages_5_15_hardened.perf (linuxKernel.packages.linux_5_15_hardened.perf)</li>
    <li>linuxPackages_5_15_hardened.systemtap (linuxKernel.packages.linux_5_15_hardened.systemtap)</li>
    <li>linuxPackages_latest.perf (linuxKernel.packages.linux_5_16.perf)</li>
    <li>linuxPackages_latest.systemtap (linuxKernel.packages.linux_5_16.systemtap)</li>
    <li>linuxKernel.packages.linux_5_4.perf</li>
    <li>linuxKernel.packages.linux_5_4.systemtap</li>
    <li>linuxPackages_5_4_hardened.perf (linuxKernel.packages.linux_5_4_hardened.perf)</li>
    <li>linuxPackages_5_4_hardened.systemtap (linuxKernel.packages.linux_5_4_hardened.systemtap)</li>
    <li>linuxPackages_latest-libre.perf (linuxKernel.packages.linux_latest_libre.perf)</li>
    <li>linuxPackages_latest-libre.systemtap (linuxKernel.packages.linux_latest_libre.systemtap)</li>
    <li>linuxPackages-libre.perf (linuxKernel.packages.linux_libre.perf)</li>
    <li>linuxPackages-libre.systemtap (linuxKernel.packages.linux_libre.systemtap)</li>
    <li>linuxPackages_lqx.perf (linuxKernel.packages.linux_lqx.perf)</li>
    <li>linuxPackages_lqx.systemtap (linuxKernel.packages.linux_lqx.systemtap)</li>
    <li>linuxPackages_testing_bcachefs.perf (linuxKernel.packages.linux_testing_bcachefs.perf)</li>
    <li>linuxPackages_testing_bcachefs.systemtap (linuxKernel.packages.linux_testing_bcachefs.systemtap)</li>
    <li>linuxPackages_xanmod.perf (linuxKernel.packages.linux_xanmod.perf)</li>
    <li>linuxPackages_xanmod.systemtap (linuxKernel.packages.linux_xanmod.systemtap)</li>
    <li>linuxPackages_zen.perf (linuxKernel.packages.linux_zen.perf)</li>
    <li>linuxPackages_zen.systemtap (linuxKernel.packages.linux_zen.systemtap)</li>
    <li>python310Packages.rasterio</li>
    <li>python39Packages.cartopy</li>
    <li>python39Packages.fiona</li>
    <li>python39Packages.geopandas</li>
    <li>python39Packages.osmnx</li>
    <li>python39Packages.rasterio</li>
    <li>qtcreator</li>
    <li>quickder</li>
    <li>saga</li>
    <li>vigra</li>
  </ul>
</details>

**Comment by RamKromberg** [comment]
While I still often use [metamorphose2](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/misc/metamorphose2/default.nix), I can make due with scripts and such so feel free to drop it if it's holding back the python2 deprecation and no one else raises objections.

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -301,6 +305,7 @@ mapAliases ({
   gdb-multitarget = gdb; # added 2017-11-13
   gdk_pixbuf = gdk-pixbuf; # added 2019-05-22
   gettextWithExpat = gettext; # 2016-02-19
+  getmail = throw "getmail was has been removed from nixpkgs, migrate to getmail6."; # added 2022-01-12
```
Comment (dotlambda) [inline pkgs/top-level/aliases.nix:308]:
```suggestion
  getmail = throw "getmail has been removed from nixpkgs, migrate to getmail6."; # added 2022-01-12
```

**Review by dotlambda — COMMENTED** [review]
> drop gdal_2 pin 

Let's do the same for fiona.

**Comment by mweinelt** [comment]
@dotlambda Done.

**Comment by dotlambda** [comment]
@mweinelt Opae doesn't build. How did you test that it works with Python 3?

**Comment by mweinelt** [comment]
Mentioned in https://github.com/NixOS/nixpkgs/pull/154697#issuecomment-1010596920.

Previously:
```
❯ nix-build -A opae
error: pytest-forked-1.4.0 not supported for interpreter python2.7
```

So we don't loose anything either way.

**Comment by dotlambda** [comment]
> So we don't loose anything either way.

But if it doesn't work Python 3 (we don't know whether it does) it should be removed instead.

**Comment by mweinelt** [comment]
It is maintained upstream and probably just needs a maintainer update.

https://github.com/OPAE/opae-sdk

**Comment by jbedo** [comment]
I'm ok with removing lumpy, python 3 support has been an open issue since 2016 https://github.com/arq5x/lumpy-sv/issues/124.

**Review by dotlambda — APPROVED** [review]

**Pull request merged**